File: codes/btq13/bac.py
#The implementation and plotting scripts are adapted from: 
#{https://github.com/Chamarthikrishnamadhur/ID1063/tree/master/07-09-26/BACTERIA}


import math
import matplotlib.pyplot as plt
import numpy as np 
x1=np.linspace(0,0.2,20)
y1=3*np.exp(math.log(2)*x1)-2
x=np.loadtxt("x.dat") #taking input from x.dat file 
y=np.loadtxt("y.dat") #taking input of y.dat file 
plt.plot(x1,y1,color='r')
plt.stem(x,y) #stem plot
plt.savefig("bac.png")

File: codes/btq13/bac.c
// The implementation and plotting scripts are adapted from: 
// {https://github.com/Chamarthikrishnamadhur/ID1063/tree/master/07-09-26/BACTERIA}
#include<stdio.h>
#include<math.h>
double euler(double y){
	return (y+0.01*(log(2)*(y+2))); //using given expression
}
int main(){
	double y=1;
	double h=0.01;
	double x=0;
	FILE *fp1=fopen("y.dat","w"); //creating dat files for plotting in python
	FILE *fp2=fopen("x.dat","w");
	for (int i=0;i<200;i++){	
	fprintf(fp1,"%lf\n",euler(y));
	y=euler(y);
	fprintf(fp2,"%lf\n",x);
	x+=h;

	}
}

